lnwallet/channel.go

```diff
@@ -3702,36 +3861,34 @@ func (lc *LightningChannel) ReceiveRevocation(revMsg *lnwire.RevokeAndAck) ([]*P
 		lc.remoteCommitChain.tail().height,
 		lc.remoteCommitChain.tail().height+1)
 
-	// At this point, the revocation has been accepted, and we've rotated
-	// the current revocation key+hash for the remote party. Therefore we
-	// sync now to ensure the revocation producer state is consistent with
-	// the current commitment height and also to advance the on-disk
-	// commitment chain.
-	if err := lc.channelState.AdvanceCommitChainTail(); err != nil {
-		return nil, err
-	}
+	remoteChainTail := lc.remoteCommitChain.tail().height + 1
```

Comment:
What is the +1 for?

Reply:
This will be the height of the remote chain after the update happens, I've added a comment to describe this.

Before, the payment descriptors were gathered after the change was written to disk, so the height would have been proplery updated. With the inclusion of forwarding packages, we have to do the gather pre-update, so the plus one ensures we are using the same height as before